GDJS/Runtime/events-tools/networktools.js

```diff
@@ -149,26 +149,40 @@ gdjs.evtTools.network.objectVariableStructureToJSON = function (
 };
 
 gdjs.evtTools.network._objectToVariable = function (obj, variable) {
-  if (obj == null) {
+  if (obj === null) {
     variable.setString("null");
   } else if (!isNaN(obj)) {
-    //Number
     variable.setNumber(obj);
   } else if (typeof obj == 'string' || obj instanceof String) {
     variable.setString(obj);
+  } else if (typeof obj === "undefined") {
+    // Do not modify the variable, as there is no value to set it to.
```

Comment:
What about something like:

```
{
  firstName: 'test'
  lastName: undefined
}
```

Will this get properly converted? (the issue is that JS is mixing "undefined" as in "not even declared" and "undefined" as in "a special value that means something was not set".

Reply:
Well, I assumed that you would use null for a special value that wasn't set. Usually, when you set something to undefined, it is to "delete" it without creating a new hidden class. Therefore I assume you wouldn't want the value to be converted into the variable at all.

Reply:
In GDevelop that would be true, in the outside world (i.e: a payload coming from a server), not sure if this holds. But let's keep it like this, having something as undefined can justify to not have it reflected in the variable :)